File: lang/fib/fib.js
#!/usr/bin/node

if(process.argv.length != 3)
    process.exit()

var n = process.argv[2];
var a = 1;
var b = 0;
var SZ = Math.pow(2,32);

for(var i = 0; i < n; i++) {
    t = b;
    b = (a+b)%SZ;
    a = t;
}

console.log(b);

File: lang/fib/fib.py
#!/usr/bin/python
from sys import argv

def fib(n):
    SZ = 2**63

    i = 0
    a, b = 1, 0
    while i < n:
        t = b
        b = (a+b) & 0x7fffffffffffffff;
        a = t
        i += 1

    return b

if __name__ == '__main__':
    print fib(int(argv[1]))
